[ADD] New module partner_multi_company

oihane opened the pull request.

oihane committed:
[ADD] New module partner_multi_company

pedrobaeza commented on partner_multi_company/README.rst:
```diff
@@ -0,0 +1,64 @@
+Installation
+============
+
+This module uses the post and uninstall hooks for updating default product
```
s/product/partner

pedrobaeza commented on partner_multi_company/README.rst:
```diff
@@ -0,0 +1,64 @@
+Usage
+=====
+
+On the product form view, go to the "Information" tab, and put the companies
```
Change instructions for referring partners, not products

pedrobaeza commented on partner_multi_company/__openerp__.py:
```diff
@@ -0,0 +1,22 @@
+    "version": "8.0.1.0.0",
+    "license": "AGPL-3",
+    "depends": ["base"],
+    "author": "Odoo Community Association (OCA)",
```
Put "Serv. Tecnol. Avanzados - Pedro M. Baeza" and you if you want.

pedrobaeza commented on partner_multi_company/__openerp__.py:
```diff
@@ -0,0 +1,22 @@
+    "contributors": [
+        "Oihane Crucelaegui <oihanecruce@avanzosc.es>",
+    ],
+    "category": "Custom Module",
```
Isn't any other category?

pedrobaeza commented on partner_multi_company/hooks.py:
```diff
@@ -0,0 +1,38 @@
+
+
+def uninstall_hook(cr, registry):
+    """Restore product rule to base value."""
```
s/product/partner

pedrobaeza committed:
[FIX] partner_multi_company: Tests + README

pedrobaeza commented:
Now it's working.

By my part, :+1:

JayVora-SerpentCS commented on partner_multi_company/models/res_partner.py:
```diff
@@ -0,0 +1,66 @@
+
+    @api.multi
+    @api.depends('company_ids')
+    def _compute_company_id(self):
```
@api.one and @api.depends would be best, isn't it?

pedrobaeza replied:
`@api.one` is deprecated in v9, so I'm avoiding to use it directly. It's only one loop less...

pedrobaeza committed:
[IMP] partner_multi_company: Sync company_ids field

pedrobaeza committed:
[FIX] product_multi_company: Avoid errors in other tests

This fix prevents errors when there are other tests that create partners
and base_suspend_security is still not ready. There can be still problems
if these partners are not created with SUPERUSER.

antoniocanovas commented:
Any problem with this module?
Functionalty 👍

pedrobaeza commented:
Merging this old one that has 2 approvals.
